src/middlewared/middlewared/plugins/gluster_linux/volume.py

```diff
@@ -0,0 +1,542 @@
+        Str('volname', required=True),
+        List('volbricks', items=[
+            Dict(
+                'brick',
+                Str('peer_name', required=True),
+                Str('peer_path', required=True),
+            ),
+        ], required=True),
+        Int('replica'),
+        Int('stripe'),
+        Int('arbiter'),
+        Int('disperse'),
+        Int('disperse_data'),
+        Int('redundancy'),
+        Bool('force'),
+    ))
+    @job(lock='glustervolume_create')
+    def do_create(self, job, data):
+        """
+        Create a gluster volume.
+
+        `volname` Name to be given to the gluster volume
+        `volbricks` List of brick paths
+            `peer_name` IP or DNS name of the peer.
+            `peer_path` The full path of the brick
+
+        `replica` Number of replica bricks
+        `stripe` Number of stripe bricks
+        `arbiter` Number of arbiter bricks
+        `disperse` Number of disperse bricks
+        `disperse_data` Number of disperse data bricks
+        `redundancy` Number of redundancy bricks
+        `force` Create volume forcefully, ignoring potential warnings
+        """
+
+        self.common_validation(job)
+
+        return self.create_volume(data)
+
+    @accepts(Dict(
+        'glustervolume_start',
+        Str('volname', required=True),
+        Bool('force'),
+    ))
+    @job(lock='glustervolume_start')
+    def start(self, job, data):
+        """
+        Start a gluster volume.
+
+        `volname` Name of gluster volume
+        `force` Forcefully start the gluster volume
+        """
+
+        self.common_validation(job)
+
+        result = self.start_volume(data)
+
+        return result
+
+    @accepts(Dict(
+        'glustervolume_restart',
+        Str('volname', required=True),
+        Bool('force'),
+    ))
+    @job(lock='glustervolume_restart')
+    def restart(self, job, data):
+        """
+        Restart a gluster volume.
+
+        `volname` Name of gluster volume
+        `force` Forcefully restart the gluster volume
+        """
+
+        self.common_validation(job)
+
+        result = self.restart_volume(data)
+
+        return result
+
+    @accepts(Dict(
+        'glustervolume_stop',
+        Str('volname', required=True),
+        Bool('force'),
+    ))
+    @job(lock='glustervolume_stop')
+    def stop(self, job, data):
+        """
+        Stop a gluster volume.
+
+        `volname` Name of gluster volume
+        `force` Forcefully stop the gluster volume
+        """
+
+        self.common_validation(job)
+
+        result = self.stop_volume(data)
+
+        return result
+
+    @accepts(Dict(
+        'glustervolume_delete',
+        Str('volname', required=True),
+    ))
+    @job(lock='glustervolume_delete')
+    def do_delete(self, job, data):
+        """
+        Delete a gluster volume.
+
+        `volname` Name of the volume to be deleted
+        """
+
+        self.common_validation(job)
+
+        result = self.delete_volume(data)
```

Comment:
Also not sure how much it buys us to have methods like that, which all it does is call the wrapper.

Why not do

```
result = self.__volume_wrapper(volume.delete, data['volname'])
```
here

Reply:
Agree 100%, fixed.